examples/exports_memory.rs

```diff
@@ -0,0 +1,135 @@
+use wasmer_compiler_cranelift::Cranelift;
+use wasmer_engine_jit::JIT;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Let's declare the Wasm module with the text representation.
+    let wasm_bytes = wat2wasm(
+        br#"
+(module
+  (memory (export "mem") 1)
+
+  (global $offset i32 (i32.const 42))
+  (global $length (mut i32) (i32.const 13))
+
+  (func (export "store") (param i32) (param i32)
+    (global.set $length (i32.add (i32.const 1) (i32.sub (local.get 0) (global.get $offset))))
+    (i32.store8 (local.get 0) (local.get 1)))
+
+  (func (export "load") (result i32 i32)
+    global.get $offset
+    global.get $length)
+
+  (data (global.get $offset) "Hello, World!"))
+"#,
+    )?;
+
+    // Create a Store.
+    // Note that we don't need to specify the engine/compiler if we want to use
+    // the default provided by Wasmer.
+    // You can use `Store::default()` for that.
+    let store = Store::new(&JIT::new(&Cranelift::default()).engine());
+
+    println!("Compiling module...");
+    // Let's compile the Wasm module.
+    let module = Module::new(&store, wasm_bytes)?;
+
+    // Create an empty import object.
+    let import_object = imports! {};
+
+    println!("Instantiating module...");
+    // Let's instantiate the Wasm module.
+    let instance = Instance::new(&module, &import_object)?;
+
+    let store = instance
+        .exports
+        .get_native_function::<(i32, i32), ()>("store")?;
+    let load = instance
+        .exports
+        .get_native_function::<(), (i32, i32)>("load")?;
+
+    // Here we go.
+    //
+    // The Wasm module exports a memory under "mem". Let's get it.
+    let memory = instance.exports.get_memory("mem")?;
+
+    // Now that we have the exported memory, let's get some
+    // information about it.
+    //
+    // The first thing we might be intersted in is the size of the memory.
+    // Let's get it!
+    println!("Memory size (pages) {:?}", memory.size());
+    println!("Memory size (bytes) {:?}", memory.data_size());
+
+    // Next, we'll want to read the contents of the memory.
+    //
+    // To do so, we have to get a `View` of the memory.
+    let view = memory.view::<u8>();
+
+    // Oh! Wait, before reading the contents, we need to know
+    // where to find what we are looking for.
+    //
+    // Fortunately, the Wasm module exports a `load` function
+    // which will tell us the offset and length of the string.
+    let (offset, length) = load.call()?;
+    println!("String offset: {:?}", offset);
+    println!("String length: {:?}", length);
+
+    // We now know where to fin our string, let's read it.
+    //
+    // We will get bytes out of the memory so we need to
+    // decode them into a string.
+    let iter = view[offset as usize..(offset + length) as usize]
+        .iter()
+        .map(Cell::get);
+    let bytes: Vec<u8> = iter.collect();
+    println!("Memory contents: {:?}", str::from_utf8(&bytes)?);
```

Comment:
Then we can just do
```Rust
let (ptr, length) = load.call()?;
println!("String offset: {:?}", ptr.offset());
println!("String length: {:?}", length);

let wasm_str = ptr.get_utf8_string(memory, length).unwrap();
println!("Memory contents: {}", wasm_str); 
```

https://docs.rs/wasmer/1.0.0-alpha4/wasmer/struct.WasmPtr.html#method.get_utf8_string